frame/nfts/src/features/lock.rs

```diff
@@ -0,0 +1,120 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::*;
+use frame_support::pallet_prelude::*;
+
+impl<T: Config<I>, I: 'static> Pallet<T, I> {
+	pub(crate) fn do_lock_collection(
+		origin: T::AccountId,
+		collection: T::CollectionId,
+		lock_config: CollectionConfig,
+	) -> DispatchResult {
+		let details =
+			Collection::<T, I>::get(&collection).ok_or(Error::<T, I>::UnknownCollection)?;
+		ensure!(origin == details.freezer, Error::<T, I>::NoPermission);
+
+		CollectionConfigOf::<T, I>::try_mutate(collection, |maybe_config| {
+			let config = maybe_config.as_mut().ok_or(Error::<T, I>::NoConfig)?;
+
+			if lock_config.has_disabled_setting(CollectionSetting::TransferableItems) {
+				config.disable_setting(CollectionSetting::TransferableItems);
+			}
+			if lock_config.has_disabled_setting(CollectionSetting::UnlockedMetadata) {
+				config.disable_setting(CollectionSetting::UnlockedMetadata);
+			}
+			if lock_config.has_disabled_setting(CollectionSetting::UnlockedAttributes) {
+				config.disable_setting(CollectionSetting::UnlockedAttributes);
+			}
```

Comment:
I'm checking only those 3 settings since the last one (`DepositRequired`) can be set by the root only. In the future, I expect more settings with custom business logic or gated control

Reply:
> With bit flags something config &= !lock_config should work. Or more Rusty and less Cy: config -= lock_config.

Wouldn't that significantly reduce the code readability? Giles wrote above that we should write a more readable code, that's why I added `has_disabled_setting()`, `disable_setting()` and other helpers